contracts/sol6/Dao/KyberStaking.sol

```diff
@@ -39,47 +37,17 @@ contract KyberStaking is IKyberStaking, EpochUtils, ReentrancyGuard {
         address _kncToken,
         uint256 _epochPeriod,
         uint256 _startTimestamp,
-        address _daoContractSetter
```

Comment:
we can directly take this parameters as IKyberDAO
and above for knc
IERC20
And avoid the casting later.

 I think this is what we usually do.